skbio/table/_augment.py

```diff
@@ -0,0 +1,560 @@
+
+        Returns
+        -------
+        augmented_matrix : numpy.ndarray
+            The augmented matrix.
+        augmented_label : numpy.ndarray
+            The augmented label, in one-hot encoding.
+
+        Examples
+        --------
+        >>> from skbio.table import Table
+        >>> from skbio.table import Augmentation
+        >>> data = np.arange(40).reshape(10, 4)
+        >>> sample_ids = ['S%d' % i for i in range(4)]
+        >>> observ_ids = ['O%d' % i for i in range(10)]
+        >>> table = Table(data, observ_ids, sample_ids)
+        >>> table_compositional = table.norm(axis="sample")
+        >>> label = np.random.randint(0, 2, size=4)
+        >>> augmentation = Augmentation(table_compositional, label)
+        >>> aug_matrix, aug_label = augmentation.aitchison_mixup(n_samples=5)
+        >>> print(aug_matrix.shape)
+        (10, 9)
+        >>> print(aug_label.shape)
+        (9, 2)
+
+        Notes
+        -----
+        The algorithm is based on [1]_, and leverages the Aitchison geometry
+        to guide data augmentation in compositional microbiome data,
+        this is essentially the vanilla mixup in the Aitchison geometry.
+        This mixup method only works on the Compositional data.
+        where a set of datapoints are living in the simplex:
+        :math:`x_i > 0`, and :math:`\sum_{i=1}^{p} x_i = 1`.
+        The augmented sample is computed as the linear combination of
+        the two samples in the Aitchison geometry. In the Aitchision
+        Geometry, we define the addition and scalar multiplication as:
+
+        .. math::
+
+            \lambda \otimes s =
+            \frac{1}{\sum_{j=1}^{p} s_j^{\lambda}}
+            (x_1^{\lambda}, x_2^{\lambda}, ..., x_p^{\lambda})
+
+        .. math::
+
+            s \oplus t =
+            \frac{1}{\sum_{j=1}^{p} s_j t_j}
+            (s_1 t_1, s_2 t_2, ..., s_p t_p)
+
+        .. math::
+
+            s = (\lambda \otimes  s_1) \oplus ((1 - \lambda) \otimes s_2)
+
+        The label is computed as the linear combination of
+        the two labels of the two samples
+
+        .. math::
+
+            y = \lambda \cdot y_1 + (1 - \lambda) \cdot y_2
+
+        By mixing the counts of two samples, Aitchison mixup preserves the
+        compositional nature of the data, and the sum-to-one property.
+
+        References
+        ----------
+        .. [1] Gordon-Rodriguez, E., Quinn, T., & Cunningham, J. P. (2022).
+            Data augmentation for compositional data: Advancing predictive
+            models of the microbiome. Advances in Neural Information Processing
+            Systems, 35, 20551-20565.
+
+        """
+        if not np.allclose(np.sum(self.matrix, axis=0), 1):
+            warnings.warn("The data is not compositional, it will be normalized first.")
+            if np.any(np.sum(self.matrix, axis=0) == 0):
+                self.matrix = self.matrix + 1e-5
+            self.matrix = self.matrix / np.sum(self.matrix, axis=0)
```

Comment:
scikit-bio has a function [`closure`](https://scikit.bio/docs/latest/generated/skbio.stats.composition.closure.html) for this. It doesn't require an arbitrary pseudocount like 1e-5. Use it with caution though.